boards/beagle/beaglebone_ai64/doc/index.rst

```diff
@@ -0,0 +1,112 @@
+Main domain.
+
+L1 Memory System
+----------------
+
+* 16 KB instruction cache.
+* 16 KB data cache.
+* 64 KB TCM.
+
+Region Address Translation
+--------------------------
+
+The RAT module performs a region based address translation. It translates a
+32-bit input address into a 48-bit output address. Any input transaction that
+starts inside of a programmed region will have its address translated, if the
+region is enabled.
+
+VIM Interrupt Controller
+------------------------
+
+The VIM aggregates device interrupts and sends them to the R5F CPU(s). The VIM
+module supports 512 interrupt inputs per R5F core. Each interrupt can be either
+a level or a pulse (both active-high). The VIM has two interrupt outputs per core
+IRQ and FIQ.
+
+Supported Features
+******************
+
+The board configuration supports,
+
++-----------+------------+-----------------------+
+| Interface | Controller | Driver/Component      |
++===========+============+=======================+
+| UART      | on-chip    | serial port-polling   |
+|           |            | serial port-interrupt |
++-----------+------------+-----------------------+
+
+Other hardwares features are currently not supported.
+
+Running Zephyr
+**************
+
+The J721E does not have a separate flash for the R5 cores. Because of this
+the A72 core has to load the program for the R5 cores to the right memory
+address, set the PC and start the processor.
+This can be done from Linux on the A72 core via remoteproc.
+
+This is the memory mapping from A72 to the memory usable by the R5. Note that
+the R5 cores always see their local ATCM at address 0x00000000 and their BTCM
+at address 0x41010000.
+
++------------+--------------+--------------+--------------+--------------+--------+
+| Region     | R5FSS0 Core0 | R5FSS0 Core1 | R5FSS1 Core0 | R5FSS1 Core1 | Size   |
++============+==============+==============+==============+==============+========+
+| ATCM       | 0x05c00000   | 0x05d00000   | 0x05e00000   | 0x05f00000   | 32KB   |
++------------+--------------+--------------+--------------+--------------+--------+
+| BTCM       | 0x05c10000   | 0x05d10000   | 0x05e10000   | 0x05f00000   | 32KB   |
++------------+--------------+--------------+--------------+--------------+--------+
+| DDR0       | 0xA2000000   | 0xA3000000   | 0xA4000000   | 0xA5000000   | 1MB    |
++------------+--------------+--------------+--------------+--------------+--------+
+| DDR1       | 0xA2100000   | 0xA3000000   | 0xA4100000   | 0xA5000000   | 15MB   |
++------------+--------------+--------------+--------------+--------------+--------+
+
+Steps to run the image
+----------------------
+
+The example shows how to load an image on Cortex R5FSS0_CORE0 on J721e.
+
+| Copy Zephyr image to the /lib/firmware/ directory.
+| ``cp build/zephyr/zephyr.elf /lib/firmware/``
+|
+| Ensure the Core is not running.
+| ``echo stop > /sys/class/remoteproc/remoteproc18/state``
```

Comment:
The `remoteproc18` is not hard-coded, as in, it is assigned based on configuration order during boot and might change between Linux kernel, device-tree and other Linux system changes. We've been trying to setup a set of udev rules that provide symlinks to absolute paths based on the hardware core. Please try to use those to provide documentation that is accurate over a longer term.

Reply:
I'm assuming [these are the latest names](https://github.com/beagleboard/repos-arm64/blob/main/bb-customizations/suite/trixie/debian/86-remoteproc-noroot.rules)? So that would mean for this `j7-main-r5f0_0`. Seems reasonable, I'll update.

(I do have a couple issues with the names though, but I can take that up later on the BeagleBoard forums/discord)

Reply:
Seems good now.